pkg/util/containers/metrics/cgroup_detect.go

```diff
@@ -24,7 +24,7 @@ import (
 )
 
 var (
-	containerRe = regexp.MustCompile("[0-9a-f]{64}")
+	containerRe = regexp.MustCompile("[0-9a-f]{64}|[0-9a-f]{8}(-[0-9a-f]{4}){4}")
```

Comment:
could you please add a comment to explain why we extended this? could you also add some test cases for garden containers in `cgroup_detect_test.go` ?